File: paddler_client_python/paddler_client/continue_from_conversation_history_params.py
from pydantic import BaseModel

from paddler_client.conversation_message import ConversationMessage
from paddler_client.grammar_constraint import GrammarConstraint
from paddler_client.tool import Tool


class ContinueFromConversationHistoryParams(BaseModel):
    add_generation_prompt: bool
    conversation_history: list[ConversationMessage]
    enable_thinking: bool
    grammar: GrammarConstraint | None = None
    max_tokens: int
    tools: list[Tool] = []

File: paddler_client_javascript/src/schemas/ContinueFromConversationHistoryParams.ts
import { z } from "zod";

import { ConversationMessageSchema } from "./ConversationMessage";
import { GrammarConstraintSchema } from "./GrammarConstraint";
import { ToolSchema } from "./Tool";

export const ContinueFromConversationHistoryParamsSchema = z
  .object({
    add_generation_prompt: z.boolean(),
    conversation_history: z.array(ConversationMessageSchema),
    enable_thinking: z.boolean(),
    grammar: GrammarConstraintSchema.nullable().optional(),
    max_tokens: z.number().int(),
    parse_tool_calls: z.boolean().optional(),
    tools: z.array(ToolSchema).optional(),
  })
  .strict();

export type ContinueFromConversationHistoryParams = z.infer<
  typeof ContinueFromConversationHistoryParamsSchema
>;
